app/src/main/java/com/igordotsenko/dotsenkorssreader/AddChannelFragment.java

```diff
@@ -163,8 +167,94 @@ protected String doInBackground(String... params) {
         protected void onPostExecute(String result) {
             super.onPostExecute(result);
             progressDialog.dismiss();
-            activity.updateChannelList();
-            Toast.makeText(activity, result, Toast.LENGTH_SHORT).show();
+            Toast.makeText(mActivity, result, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private Channel insertIntoChannel(Channel channel) {
+        long id = getLastChannelId() + 1;
+        channel.setId(id);
+
+        ContentValues cv = new ContentValues();
+
+        cv.put(Channel.ID, id);
+        cv.put(Channel.TITLE, channel.getTitle());
+        cv.put(Channel.LINK, channel.getLink());
+        cv.put(Channel.LAST_BUILD_DATE, channel.getLastBuildDate());
+        cv.put(Channel.LAST_BUILD_DATE_LONG, channel.getLastBuildDateLong());
+
+        mActivity.getContentResolver().insert(
+                ReaderContentProvider.ContractClass.CHANNEL_CONTENT_URI, cv);
+
+        return channel;
+    }
+
+    private boolean channelIsAlreadyAdded(String url){
+        String selection = Channel.LINK + " = ?";
+        String[] selectionArgs = { url };
+
+        Cursor cursor = mActivity.getContentResolver().query(
+                ReaderContentProvider.ContractClass.CHANNEL_CONTENT_URI,
+                null, selection, selectionArgs, null);
+
+        // If records exists - cursor has more than 0 rows
+        boolean recordExists = cursor.getCount() > 0;
+
+        cursor.close();
+
+        return recordExists;
+    }
+
+    private boolean channelIsAlreadyAdded(Channel channel){
+        String selection = Channel.TITLE + " = ?";
```

Comment:
Both 'channelIsAlreadyAdded' methods looks 90% same. Think about how to extract the common logic, so there will be no duplicated code)